plugins/modules/relationship_association.py

```diff
@@ -0,0 +1,151 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: relationship_association
+short_description: Creates or removes a relationship association from Nautobot
+description:
+  - Creates or removes a relationship association from Nautobot
+author:
+  - Network to Code (@networktocode)
+  - Joe Wesch (@joewesch)
+requirements:
+  - pynautobot
+version_added: "4.0.0"
+options:
+  url:
+    description:
+      - URL of the Nautobot instance resolvable by Ansible control host
+    required: true
+    type: str
+  token:
+    description:
+      - The token created within Nautobot to authorize API access
+    required: true
+    type: str
+  relationship:
+    description:
+      - The Relationship UUID to add the association to
+    required: true
+    type: str
+  source_type:
+    description:
+      - The app_label.model for the source of the relationship
+    required: true
+    type: str
+  source_id:
+    description:
+      - The UUID of the source of the relationship
+    required: true
+    type: str
+  destination_type:
+    description:
+      - The app_label.model for the destination of the relationship
+    required: true
+    type: str
+  destination_id:
+    description:
+      - The UUID of the destination of the relationship
+    required: true
+    type: str
+  query_params:
+    description:
+      - This can be used to override the specified values in ALLOWED_QUERY_PARAMS that is defined
+      - in plugins/module_utils/utils.py and provides control to users on what may make
+      - an object unique in their environment.
+    required: false
+    type: list
+    elements: str
+  state:
+    description:
+      - Use C(present) or C(absent) for adding or removing.
+    choices: [ absent, present ]
+    default: present
+    type: str
+  validate_certs:
+    description:
+      - |
+        If C(no), SSL certificates will not be validated.
+        This should only be used on personally controlled sites using self-signed certificates.
+    default: true
+    type: raw
+"""
+
+EXAMPLES = r"""
+- name: "Test relationship association creation/deletion"
+  connection: local
+  hosts: localhost
+  gather_facts: False
+  tasks:
+    - name: Create relationship association
+      networktocode.nautobot.relationship_association:
+        url: http://nautobot.local
+        token: thisIsMyToken
+        relationship: 01234567-abcd-0123-abcd-012345678901
+        source_type: dcim.device
+        source_id: abcdefgh-0123-abcd-0123-abcdefghijkl
+        destination_type: ipam.vrf
+        destination_id: 01234567-abcd-0123-abcd-123456789012
+
+    - name: Delete relationship association
+      networktocode.nautobot.relationship_association:
+        url: http://nautobot.local
+        token: thisIsMyToken
+        relationship: 01234567-abcd-0123-abcd-012345678901
+        source_type: dcim.device
+        source_id: abcdefgh-0123-abcd-0123-abcdefghijkl
```

Comment:
What we could potentially do to make it easier for the user is to find the ID for them by taking the source.type and kwargs under `source` and find it for them, but it is extra logic I don't know if we have currently.

Reply:
The source/destination ID is not the ID of the type, but rather the actual device or IP Address object to relate to the other.

Reply:
Correct. Here is an example:

```
    - name: Create relationship association
      networktocode.nautobot.relationship_association:
        url: http://nautobot.local
        token: thisIsMyToken
        relationship: 01234567-abcd-0123-abcd-012345678901
        source_type: dcim.device
        source:
          name: den-edge-01
          site: DEN
        destination_type: ipam.vrf
        destination_id: 01234567-abcd-0123-abcd-123456789012
```
We can split on the **source_type** that will provide us the correct app and endpoint to query against.

We would then search for that for them rather than them having to lookup the ID by using the lookup plugin.

Reply:
Ah, yes. There are many places convenience logic like that could be added for sure.